src/Battlescape/SkillMenuState.cpp

```diff
@@ -73,25 +67,37 @@ SkillMenuState::SkillMenuState(BattleAction *action, int x, int y) : ActionMenuS
 		Options::keyBattleActionItem2,
 		Options::keyBattleActionItem1
 	};
-	auto soldier = _action->actor->getGeoscapeSoldier();
+
 	for (auto skill : soldier->getRules()->getSkills())
 	{
-		if (!hotkeys.empty()
-			&& soldierHasAllRequiredBonusesForSkill(soldier, skill)
+		if (soldierHasAllRequiredBonusesForSkill(soldier, skill)
```

Comment:
we need break condition for `id == std::size(_actionMenu)`

Reply:
I don't get this, what do you mean here?